xbmc/utils/DiscsUtils.cpp

```diff
@@ -66,3 +75,25 @@ UTILS::DISCS::DiscInfo UTILS::DISCS::ProbeBlurayDiscInfo(const std::string& medi
 #endif
   return info;
 }
+
+bool UTILS::DISCS::IsBlurayDiscImage(const CFileItem& item)
+{
+  if (!item.IsDiscImage())
+    return false;
+
+  static constexpr std::array<const char*, 4> blurayFiles = {
+      "index.bdmv",
+      "INDEX.BDM",
+      "BDMV/index.bdmv",
+      "BDMV/INDEX.BDM",
+  };
+  CURL url("udf://");
+  url.SetHostName(item.GetDynPath());
+
+  return std::ranges::any_of(blurayFiles,
+                             [&url](const std::string& file)
+                             {
+                               url.SetFileName(file);
+                               return XFILE::CFile::Exists(url.Get());
+                             });
```

Comment:
```suggestion
  return std::ranges::any_of(blurayFiles,
                             [&url](std::string_view file)
                             {
                               url.SetFileName(file);
                               return XFILE::CFile::Exists(url.Get());
                             });
```
If you use a `const std::string&` as the parameter here you have to construct two `std::string` per `blurayFiles` element: One to pass into the function and one copy to pass into `url.SetFileName(file)`. With the `std::string_view` parameter the first creation can be omitted.

Reply:
Looks like you have to put `std::string()` around `file`:

```
  static constexpr std::array<std::string_view, 4> blurayFiles = {
      "index.bdmv",
      "INDEX.BDM",
      "BDMV/index.bdmv",
      "BDMV/INDEX.BDM",
  };
  CURL url("udf://");
  url.SetHostName(item.GetDynPath());

  return std::ranges::any_of(blurayFiles,
                             [&url](std::string_view file)
                             {
                               url.SetFileName(std::string(file));
                               return XFILE::CFile::Exists(url.Get());
                             });
```

Reply:
It looks possible to change SetFileName to accept a string_view parameter for more versatility instead of string. It makes a copy anyway, so copy from string or string_view, same deal. Dedicated commit.
Alternatively an array of strings will require the least amount of conversions.

Reply:
Will do. An array of `std::string` doesn't work:
![image](https://github.com/user-attachments/assets/13d68f8d-99d8-4c50-a93d-daeb7ab4d0aa)
Not completely sure why though, as although a `std::string` can be variable length surely that length can be determined at complie time??

Reply:
`std::string` can be used with `constexpr`, but all memory that is allocated at compile time during constexpr evaluation must also be freed at compile time. For that reason you can't create a `std::string` at compile time and store it for usage at runtime.

In detail: https://quuxplusone.github.io/blog/2023/09/08/constexpr-string-firewall/

Reply:
Thank you, that's interesting.